corehq/motech/openmrs/models.py

```diff
@@ -118,7 +118,8 @@ class OpenmrsImporter(Document):
     column_map = ListProperty(ColumnMapping)
 
     def __str__(self):
-        return self.server_url
+        url = "@".join((self.username, self.server_url)) if self.username else self.server_url
```

Comment:
This is url on which request is sent

Reply:
Yes, it's the base URL, e.g. `https://www.example.com/openmrs/` and then the ID of the report whose data we will be importing will be appended to that. 

We now differentiate using `username` as well, because different OpenMRS users have access to different things. It is possible that the same server could be configured in CommCare with different users, so the URL wouldn't be unique and we'd need to know the username as well.

Reply:
and hitting that URL just need the username and not the password

Reply:
Oh, the `__str__()` method is just to print a string value for the importer. We don't want to print the password.

When we hit the URL, we use a Requests instance for that, and we do give Requests the URL, username and password. e.g. [We do that over here](https://github.com/dimagi/commcare-hq/blob/3e411f3914979848ca486eac94d186e9bd14fb04/corehq/motech/openmrs/tasks.py#L189).

Reply:
The [Requests class](https://github.com/dimagi/commcare-hq/blob/f3057b804c34c25d890f031eaf9a7f6419af8447/corehq/motech/requests.py#L47) is useful for a bunch of things. We use it for making API calls, and we also use it for logging and for notifying on errors.